File: angular/src/app/core/enums/age-range.ts
export enum AgeRange {
  THREE_TO_FIVE = 'THREE_TO_FIVE',
  SIX_TO_EIGHT = 'SIX_TO_EIGHT',
  NINE_TO_TWELVE = 'NINE_TO_TWELVE'
  
}

export function ageRangeLabel(value: AgeRange): string {
  switch (value) {
    case AgeRange.THREE_TO_FIVE:
      return '3–5 anos';
    case AgeRange.SIX_TO_EIGHT:
      return '6–8 anos';
    case AgeRange.NINE_TO_TWELVE:
      return '9–12 anos';
    default:
      return value;
  }
}

File: fabulary-stories-service/src/main/java/com/fabulary/stories/enums/AgeRange.java
package com.fabulary.stories.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum AgeRange {
    THREE_TO_FIVE("THREE_TO_FIVE"),
    SIX_TO_EIGHT("SIX_TO_EIGHT"),
    NINE_TO_TWELVE("NINE_TO_TWELVE");

    private final String value;

    AgeRange(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    @JsonCreator
    public static AgeRange fromValue(String value) {
        return AgeRange.valueOf(value);
    }
}
